### PR TITLE
Replace argh library with fire library.

### PR DESCRIPTION
**Pull request by briandersn**
Fire is slightly more straightforward for minigo's usage
since everything can be passed directly to the fire.Fire()
rather than indirectly through argh.add_commands().

argparse usage is also hidden and doesn't need to be
imported separately and used as a proxy.

### CODE REVIEW

File: rl_loop.py [hunk 1]
```diff
@@ -158,13 +157,15 @@ def backfill():
...
 if __name__ == '__main__':
     cloud_logging.configure()
     remaining_argv = flags.FLAGS(sys.argv, known_only=True)
-    argh.dispatch(parser, argv=remaining_argv[1:])
+    fire.Fire({
+        'train': train,
+        'selfplay': selfplay,
+        'backfill': backfill,
+        'bootstrap': bootstrap,
+        'fsdb.game_counts': fsdb.game_counts,
```
Comment (brilee) [inline rl_loop.py]: I think this one would be called as `game_counts`, not `fsdb.game_counts`.

**Comment by brilee** [comment]
Looks good other than the one imported method.

File: rl_loop.py (hunk 1 above)
Comment (briandersn) [reply rl_loop.py]: Done.

**Comment by briandersn** [comment]
It looks like the presubmit bots don't have the fire library installed. How can we update them?

**Comment by brilee** [comment]
I think if you add fire to requirements.txt, the docker image will pick up the changes.

**Comment by briandersn** [comment]
I guess updating requirements.txt doesn't work if it's part of the same patch? I'll try pulling in the fire dependency into requirements.txt as a separate patch first.

**Comment by briandersn** [comment]
Added fire to requirements.txt here: https://github.com/tensorflow/minigo/pull/449

**Pull request merged**